// react-native-paper/lib/commonjs/components/List/utils.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getRightStyles = exports.getLeftStyles = void 0;

var _reactNative = require("react-native");

const getLeftStyles = (alignToTop, description, isV3) => {
  const stylesV3 = {
    marginRight: 0,
    marginLeft: 16,
    alignSelf: alignToTop ? 'flex-start' : 'center'
  };

  if (!description) {
    return { ...styles.iconMarginLeft,
      ...styles.marginVerticalNone,
      ...(isV3 && { ...stylesV3
      })
    };
  }

  if (!isV3) {
    return styles.iconMarginLeft;
  }

  return { ...styles.iconMarginLeft,
    ...stylesV3
  };
};

exports.getLeftStyles = getLeftStyles;

const getRightStyles = (alignToTop, description, isV3) => {
  const stylesV3 = {
    marginLeft: 16,
    alignSelf: alignToTop ? 'flex-start' : 'center'
  };

  if (!description) {
    return { ...styles.iconMarginRight,
      ...styles.marginVerticalNone,
      ...(isV3 && { ...stylesV3
      })
    };
  }

  if (!isV3) {
    return styles.iconMarginRight;
  }

  return { ...styles.iconMarginRight,
    ...stylesV3
  };
};

exports.getRightStyles = getRightStyles;

const styles = _reactNative.StyleSheet.create({
  marginVerticalNone: {
    marginVertical: 0
  },
  iconMarginLeft: {
    marginLeft: 0,
    marginRight: 16
  },
  iconMarginRight: {
    marginRight: 0
  }
});
